Dockerfile

```
@@ -1,13 +1,40 @@
-FROM golang:1.18 AS build
+ARG REG=docker.io
+FROM ${REG}/library/golang:1.17 AS build
 WORKDIR /src
-ENV GO111MODULE=on
-COPY . /src
-RUN CGO_ENABLED=0 go build -o main && go test ./... -v
+RUN go install github.com/swaggo/swag/cmd/swag@v1.7.1
+COPY go.mod go.sum ./
+RUN go mod download
 
-FROM ubuntu:20.04 AS final
+COPY . .
+ARG BUILD_VERSION="local docker"
+ARG BUILD_GIT_COMMIT="HEAD"
+ARG BUILD_REF="0"
+ARG BUILD_DATE=""
+RUN chmod +x scripts/update-version.sh  \
+    && scripts/update-version.sh assets/version.yaml \
+    && make swag check \
+    && CGO_ENABLED=0 go build -o main
+
+ARG REG=docker.io
+FROM ${REG}/library/alpine:3.14 AS final
```

I don't know if there is a more current version we can use, but seems like this was copy-pasted so adding a comment here too

---

3.15 is the latest minor by now. Nice catch!

```suggestion
FROM ${REG}/library/alpine:3.15 AS final
```